src/executor/mod.rs

```diff
@@ -0,0 +1,303 @@
+    where
+        F: FnOnce() -> R + Send + 'static,
+        R: Send + 'static;
+}
+
+fn set_sync_strategy(strategy: SyncExecutor) -> Result<(), Error> {
+    COMPUTE_HEAVY_SYNC_EXECUTOR_STRATEGY
+        .set(strategy)
+        .map_err(|_| {
+            Error::AlreadyInitialized(COMPUTE_HEAVY_SYNC_EXECUTOR_STRATEGY.get().unwrap().into())
+        })?;
+
+    log::info!(
+        "initialized compute-heavy future executor strategy - {:#?}",
+        global_sync_strategy()
+    );
+
+    Ok(())
+}
+
+/// Get the currently initialized sync strategy,
+/// or the default strategy for the current feature in case no strategy has been loaded.
+///
+/// See [`SyncExecutorBuilder`] for details on strategies.
+///
+/// # Examples
+///
+/// ```
+/// use vacation::{
+///     global_sync_strategy,
+///     global_sync_strategy_builder,
+///     GlobalStrategy,
+///     ExecutorStrategy
+/// };
+///
+/// # fn run() {
+///
+/// #[cfg(feature = "tokio")]
+/// assert_eq!(global_sync_strategy(), GlobalStrategy::Default(ExecutorStrategy::SpawnBlocking));
+///
+/// #[cfg(not(feature = "tokio"))]
+/// assert_eq!(global_sync_strategy(), GlobalStrategy::Default(ExecutorStrategy::CurrentContext));
+///
+/// global_sync_strategy_builder()
+///         .initialize_current_context()
+///         .unwrap();
+///
+/// assert_eq!(global_sync_strategy(), GlobalStrategy::Initialized(ExecutorStrategy::CurrentContext));
+///
+/// # }
+/// ```
+pub fn global_sync_strategy() -> GlobalStrategy {
+    match COMPUTE_HEAVY_SYNC_EXECUTOR_STRATEGY.get() {
+        Some(strategy) => GlobalStrategy::Initialized(strategy.into()),
+        None => GlobalStrategy::Default(<&SyncExecutor>::default().into()),
+    }
+}
+
+pub(crate) fn get_global_sync_executor() -> &'static SyncExecutor {
+    COMPUTE_HEAVY_SYNC_EXECUTOR_STRATEGY
+        .get()
+        .unwrap_or_else(|| <&SyncExecutor>::default())
+}
+
+/// The stored strategy used to spawn compute-heavy futures.
+static COMPUTE_HEAVY_SYNC_EXECUTOR_STRATEGY: OnceLock<SyncExecutor> = OnceLock::new();
+
+/// The fallback strategy used in case no strategy is explicitly set
+static DEFAULT_COMPUTE_HEAVY_SYNC_EXECUTOR_STRATEGY: OnceLock<SyncExecutor> = OnceLock::new();
+
+#[non_exhaustive]
+pub(crate) enum SyncExecutor {
+    /// A non-op strategy that runs the function in the current context
+    CurrentContext(current_context::CurrentContextExecutor),
+    /// User-provided closure
+    CustomExecutor(custom_executor::CustomExecutor),
+    /// tokio task::spawn_blocking
+    #[cfg(feature = "tokio")]
+    SpawnBlocking(spawn_blocking::SpawnBlockingExecutor),
+}
+
+impl Default for &SyncExecutor {
+    fn default() -> Self {
+        DEFAULT_COMPUTE_HEAVY_SYNC_EXECUTOR_STRATEGY.get_or_init(|| {
+            log::warn!(
+                "Defaulting to CurrentContext (non-op) strategy for compute-heavy future executor"
+            );
+            SyncExecutor::CurrentContext(CurrentContextExecutor::new(None))
+        })
+    }
+}
```

Comment:
stateful default impls scare me a little bit—I wonder if we want an API more like tracing (that doesn't touch `Default` as a trait, instead relying on `init` as a convention).

Something like:

```
vacation::default_strategy::init()
```

Reply:
So you're suggesting, still keep the default stateful, just remove it from the trait? I'm fine with that, though don't fully understand the distinction.

Reply:
Took a stab at this, it still is implicitly defaulting, but no longer using the trait impl.